src/object/srv_ec_aggregate.c

```diff
@@ -0,0 +1,1205 @@
+/**
+ * (C) Copyright 2016-2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ "
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * DAOS server erasure-coded object aggregation.
+ *
+ * src/object/srv_ec_aggregate.c
+ */
+#define D_LOGFAC	DD_FAC(object)
+
+#include <stddef.h>
+#include <stdio.h>
+#include <daos/common.h>
+#include <daos_srv/vos.h>
+#include <daos_srv/daos_server.h>
+#include <daos_srv/srv_obj_ec.h>
+#include "obj_ec.h"
+#include "obj_internal.h"
+
+/* Pool/container info. */
+struct ec_agg_pool_info {
+	uuid_t		 api_pool_uuid;
+	uuid_t		 api_poh_uuid;
+	uuid_t		 api_cont_uuid;
+	uuid_t		 api_coh_uuid;
+	daos_handle_t	 api_cont_hdl;
+	uint32_t	 api_pool_version;
+	d_rank_list_t	*api_svc_list;
+	struct ds_pool	*api_pool;
+	/* Shared handle UUIDs, and service listr
+	 * are initialized in system Xstream.
+	 */
+	ABT_eventual	 api_eventual;
+};
+
+/* Parmeters used to drive iterate all.
```

Comment:
(style)  'Parmeters' may be misspelled - perhaps 'Parameters'?

Reply:
I fixed this...will be spelled correctly in the next push.